flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -1449,4 +1491,42 @@ public Void retrievePayload(ResourceID resourceID) {
             return null;
         }
     }
+
+    private class JobMasterBlocklistContext implements BlocklistContext {
+
+        @Override
+        public void blockResources(Collection<BlockedNode> blockedNodes) {
+            Map<String, Set<ResourceID>> taskManagersByNode =
+                    taskManagerToNode.entrySet().stream()
+                            .collect(
+                                    Collectors.groupingBy(
+                                            Map.Entry::getValue,
+                                            Collectors.mapping(
+                                                    Map.Entry::getKey, Collectors.toSet())));
+
+            Collection<ResourceID> blockedTaskMangers =
```

Comment:
Given that it is now iterating over all the registered taskmanagers. We can do it in a simpler and possibly more performant way, by checking all the taskmanagers to see if it is on the blocked nodes (creating a set for blocked nodes first).
Or we can maintain the `taskManagersByNode` in JobMaster to make the process even more performant.